chore: add devcontainer configuration

Garbee opened the pull request:
Provide a starting point for devcontainer use to develop lit. This can help new contributors quickly get a dev environment going in the cloud without needing to run anything locally. However, if desired a developer can run devcontainers locally in order to isolate different work environments.

This container does the following:

- Uses the existing Node 24 + TypeScript devcontainer from Microsoft as the base
- Sets up the GH CLI so it is available in case anyone is used to in their workflow
- Installs playwright dependencies on the operating system
- After a container is brought online for a workspace, like a new Codespace, `npm ci` is executed so people can get into working quickly. Then `npx playwright install` is ran so the browsers are installed with the current dependencies of the repository in scope.

It does also suggest 4 cores as a requirement. This is because on Codespaces when running build and test the CPU was maxing on the 2 core machines and triggering warnings repeatedly. At 4 cores they can show up from time-to-time, but overall the runs seem to go smoothly.

Garbee committed:
chore: add devcontainer configuration

Provide a starting point for devcontainer use to develop lit.
This can help new contributors quickly get a dev environment going
in the cloud without needing to run anything locally.

rictic reviewed (approved):
Very nice, seems very helpful!

justinfagnani commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,26 @@
+    },
+    // Install browser depenedencies for testing
+    "ghcr.io/devcontainers-contrib/features/bash-command:1": {
+      "command": "npx playwright install --with-deps"
```
Do the browsers not get installed with the playwright npm package?

Garbee replied:
So...

1. Yes, absolutely correct the browsers do get installed.
2. I need to tweak the command to install the dependencies only. I messed this up in a few earlier things I was doing until I realized the command problem.

This will become `npx playwright install-deps` in my next commit here.

justinfagnani commented:
@rictic someone needs to start the test workflows. I don't see the button on my phone though.

Garbee committed:
chore(devcontainer): fix playwright install command

Fix the playwright install command so it only installs the browser dependencies and not the browsers as well.

Garbee committed:
Change base image for development container

Garbee committed:
Merge branch 'main' into garbee/devcontainer

Garbee committed:
Set Node.js version to 24 in devcontainer.json

Updated Node.js version in devcontainer configuration.

Garbee committed:
use node base image again

Garbee committed:
install with deps

Garbee committed:
install playwright after deps
